packages/file_selector/file_selector_platform_interface/lib/src/types/x_type_group/x_type_group.dart

```diff
@@ -14,7 +15,23 @@ class XTypeGroup {
     this.mimeTypes,
     this.macUTIs,
     this.webWildCards,
-  });
+  }) {
+    _verifyExtensions();
+  }
+
+  void _verifyExtensions() {
+    if (extensions == null) return;
+    final exts = extensions!;
+    for (var i = 0; i < exts.length; i++) {
+      if (!exts[i].startsWith('.')) continue;
+      if (kDebugMode) {
+        print('extensions[${i}] with value "${exts[i]}" is invalid.'
+            ' The leading dots are being removed from the extensions'
+            ' Please fix it.');
```

Comment:
Sorry, one more nit. Looking at this again, I don't think there's really any value in telling the developer, but also fixing it; if we're going to do runtime fixup, then it's not really invalid. We should either assert (and clearly document the requirement), or just silently fix it (which I think is the better option from a developer-friendliness standpoint, since there's no ambiguity in the intent here).

So let's just remove this whole print.

Reply:
> or just silently fix it (which I think is the better option from a developer-friendliness standpoint, since there's no ambiguity in the intent here).

The only thing that concerns me is that we are mutating the extensions list. This mutation could lead to unexpected errors + hard debugging time if we don't inform them.

```dart
final extensions = await api.getExtensions();
final files = openFiles(...);
// By this time, extensions will be muted
doSomethingElseWithExtensions(extensions);
```

Reply:
Good point; we should copy the list (and document the behavior):
```

  /// Creates a new group with the given label and file extensions.
  ///
  /// A group with none of the type options provided indicates that any type is
  /// allowed.
  /// 
  /// [extensions] are normalized to a form without a leading '.'.
  XTypeGroup({
    this.label,
    List<String>? extensions,
    this.mimeTypes,
    this.macUTIs,
    this.webWildCards,
  }) : this.extensions =
            extensions == null ? null : List<String>.from(extensions);
```

Reply:
Done. PTAL